### ISSUE TITLE
String conforming to number? fails for integers

### ISSUE THREAD

**Issue by benalbrecht** [metosin/spec-tools#92]
```clojure
(spec-tools.spec/number? 1)
=> true
(spec-tools.core/conform spec-tools.spec/number? 1 spec-tools.core/string-conforming)
=> :clojure.spec.alpha/invalid
```
i think, the issue here is, that `number?` is classified as a `double` and `string->double` only checks whether the input is a `double?` before trying `Double/parseDouble`.
Since `Double/parseDouble` expects a string, an exception is thrown and `::s/invalid` is returned.

**Comment by ikitommi** [comment]
Here's the code: https://github.com/metosin/spec-tools/blob/master/src/spec_tools/conform.cljc#L26-L35

It think it should just test for `string?` and tranform, otherwise reuturn the original value and let the spec itself return the `::s/invalid`. I think it should work. Would you have time for a pull-request on this?

**Comment by benalbrecht** [comment]
i implemented your suggestion in #93, but as a result, this test fails:
```clojure
FAIL in (string->double) (conform_test.cljc:18)
expected: (= st/+invalid+ (conform/string->double _ true))
  actual: (not (= :clojure.spec.alpha/invalid true))
```
